### ISSUE TITLE
captains-log: Add config for tweaking performance (to skip accessing `err.stack`)

### ISSUE THREAD

**Issue by sailsbot** [balderdashy/sails#4035]
_From @kevinburke on June 26, 2015 0:30_

In my test environment on a near-new MBP, it takes about 100ms to log a stack trace. Here is my Logging class, instrumented:

``` javascript
var Logging;

Logging = module.exports = {
  error: function(errorObject, message) {
    var start;
    start = Date.now();
    console.log(Date.now() - start, "sails logging");
    sails.log.error(errorObject);
    return console.log(Date.now() - start, "sails done, console");
  }
};
```

I was logging an object created with `new Error('foo bar');`

Further profiling in the captains-log class revealed that accessing `err.stack` caused a 100ms pause. It might be related to the length of the stack trace, just creating an Error object with no call stack and accessing err.stack was relatively fast.

For what it's worth I observed `console.trace` had a similar penalty. Might be related to https://github.com/felixge/node-mysql/issues/439.

_Copied from original issue: balderdashy/captains-log#17_

**Comment by sailsbot** [comment]
Thanks for posting, @kevinburke.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:
- review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
- create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by sailsbot** [comment]
_From @mikermcneil on January 14, 2016 8:10_

Thanks for the insight @kevinburke.

**Comment by sailsbot** [comment]
_From @mikermcneil on January 14, 2016 8:11_

(added the "bug" label to ward off sailsbot-- she's still a little colorblind)

**Comment by sailsbot** [comment]
_From @mikermcneil on January 14, 2016 8:20_

@kevinburke were you able to tell if removing the err.stack call in captains-log contributed to a significant improvement in performance?  If so, I'll dig in there and have a closer look.  If this is causing a performance problem for you in production, it would be great to see some more details about what kind of errors are causing this (i.e. are these errors coming from Waterline or a dependency?  Is there custom `.inspect()` or `.toString()` logic in play?  That would make a lot of sense re: your point about the empty error being so much faster)

In the mean time, if you'd rather exclude stack traces from your production log files, you might take a look at the custom logger.  If we find that this is the best way to move forward for your use case in general, happy to look for another solution like a configuration flag that will inspect error instances via some other means.

**Comment by sailsbot** [comment]
_From @mikermcneil on January 14, 2016 8:21_

re: custom logger usage, see [updated docs](https://github.com/praece/sails-docs/blob/8b44895d3e62ffdea871c261372789cc164a0304/reference/sails.config/sails.config.log.md)
